### PR TITLE
Fixing processing

### PR DESCRIPTION
**Pull request by mattmolo**
Holy cow this file made me very upset. @brstew can you please run this on the dev server and make sure everything works. I didn't actually check that, so there might be a bug. But I tried to very careful, not changing all that much actually. 

If you want to see how I moved stuff around, check the commits because some of them don't change the whitespace, so you can actually see what's going on.

### CODE REVIEW

**Comment by mattmolo** [comment]
Try this link to turn off the whitespace changes: https://github.com/PurdueIEEE/boilerbooks/pull/74/files?utf8=%E2%9C%93&diff=unified&w=1

File: complete/completeprocessing.php [hunk 1]
```diff
@@ -1,146 +1,172 @@
...
+                } catch (PDOException $e) {
+                    $sqlErr =  "SQL Statement: $sql <br>";
+                    $sqlErr .= "SQL Error: " . $e->getMessage();
+
+                    $uploadErrMessage = $uploadErrMessage . "There was an error inserting the row into the database<br>";
+                    $isUploadError = true;
+                    break;
+                }
+
+                // TODO: What is this doing, also why is it here?
```
Comment (mattmolo) [inline complete/completeprocessing.php]: @rakoskr6 ?

File: complete/completeprocessing.php [hunk 2]
```diff
@@ -1,146 +1,172 @@
...
+                    $sqlErr =  "SQL Statement: $sql <br>";
+                    $sqlErr .= "SQL Error: " . $e->getMessage();
+
+                    $uploadErrMessage = $uploadErrMessage . "There was an error inserting the row into the database<br>";
+                    $isUploadError = true;
+                    break;
+                }
+
+                // TODO: What is this doing, also why is it here? 
+                $conn = null;
```
Comment (mattmolo) [inline complete/completeprocessing.php]: @rakoskr6 Can you explain this for me?

File: complete/completeprocessing.php [hunk 3]
```diff
@@ -1,146 +1,172 @@
...
+
+                    $retval = mail($to, $subject, $message, $header);
+
+                    if ($retval == true ) {
+                        //echo "Message sent successfully...";
+                    } else {
+                        //echo "Message could not be sent...";
+
+                        // TODO: Should we break here to not clear the session variables at the bottom?? idk
+                        // break?
```
Comment (mattmolo) [inline complete/completeprocessing.php]: @rakoskr6 And also this ^ (I wrote the todos, these are some questions I have)

File: complete/completeprocessing.php [hunk 4]
```diff
@@ -1,146 +1,172 @@
...
+                    }
+                }
+
+                // TODO: What is this doing? Can we clear it all the time? Only if errors? Only if success?
```
Comment (mattmolo) [inline complete/completeprocessing.php]: @rakoskr6 And lastly ^

File: complete/completeprocessing.php [hunk 5]
```diff
@@ -1,146 +1,172 @@
...
-	$completeactive = "active";
-	include '../menu.php';
+    $title = 'Boiler Books';
+    $completeactive = "active";
+    include '../menu.php';
 ?>
 
-<?php
-include '../dbinfo.php';
-echo '<div class="container">';
```
Comment (mattmolo) [inline complete/completeprocessing.php]: Lets talk about what's wrong here, first, why are we just echo-ing html.. in the code I moved the div above the `<?php` start and then we don't have to do that anymore

File: complete/completeprocessing.php [hunk 6]
```diff
@@ -1,146 +1,172 @@
...
-	$purchasedate = test_input($_POST["purchasedate"]);
-	$purchasedateorig = $purchasedate;
-
-	$purchasedate = str_replace('-','/',$purchasedate);
-	$purchasedate = date('Y-m-d H:i:s', strtotime($purchasedate));  
-	$purchasedatetemp = $purchasedate;
-	$receipt = $servername . "/";
-	$target_dir = "../receipts/";
-	$target_dir_save = "/receipts/";
-	$uploadOk = 1;
```
Comment (mattmolo) [inline complete/completeprocessing.php]: STOP USING 1's and 0's FOR TRUE/FALSE

Php has 1) weird dynamic type casting 2) has real true/false types

So don't use 1 and 0

File: complete/completeprocessing.php [hunk 7]
```diff
@@ -1,146 +1,172 @@
...
-	$target_file = $target_dir . $committee . "_" . $item . "_" . $purchaseID . "." . $FileType;
-	$target_file_save = $target_dir_save . $committee . "_" . $item . "_" . $purchaseID . "." . $FileType;
-	// Check if file already exists
-	if (file_exists($target_file)) {
-		$uploaderr = $uploaderr . " Your file already exists on the server<br>";
-		$uploadOk = 0;
-	}
-	// Check file size
-	// file size must be less than 2MB
-	if ($_FILES["fileToUpload"]["size"] > 2048000 || $_FILES["fileToUpload"]["size"] == 0) {
```
Comment (mattmolo) [inline complete/completeprocessing.php]: If the file size is 0 the error message is wrong, so I split that up

Comment (mattmolo) [reply complete/completeprocessing.php]: Also easier to make sure your zero's are correct with some math:
`2 * 1024 * 1024`

File: complete/completeprocessing.php [hunk 8]
```diff
@@ -1,146 +1,172 @@
...
-		$uploaderr = $uploaderr . "Your reimbursement cert is larger than 2MB<br>";
-		$uploadOk = 0;
-	}
-	// Allow certain file formats
-	if($FileType != "pdf" and $FileType != "PDF" and $FileType != "jpg" and $FileType != "jpeg"and $FileType != "JPG"and $FileType != "JPEG") {
```
Comment (mattmolo) [inline complete/completeprocessing.php]: Converted to putting the correct file types in an array, and using strtolower making it much easier to maintain and doesn't worry about weird capitalization.

File: complete/completeprocessing.php [hunk 9]
```diff
@@ -1,146 +1,172 @@
...
-		$uploaderr = $uploaderr . "Only PDFs and JPEGs are allowed<br>";
-		//echo $FileType;
-		$uploadOk = 0;
```
Comment (mattmolo) [inline complete/completeprocessing.php]: stop using 0 :P

File: complete/completeprocessing.php [hunk 10]
```diff
@@ -1,146 +1,172 @@
...
-			    $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password);
-			    // set the PDO error mode to exception
-			    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
-			    $sql = "UPDATE Purchases SET modifydate = NOW(), purchasedate='$purchasedate', cost='$cost', status='Purchased', comments='$comments',
-				receipt='$receipt' WHERE Purchases.purchaseID = '$purchaseID'";
-				// use exec() because no results are returned
-			    $conn->exec($sql);
-			    //echo "New record created successfully";
-			    //echo $sql;
-			    }
```
Comment (mattmolo) [inline complete/completeprocessing.php]: 1) wtf is this indentation

File: complete/completeprocessing.php [hunk 11]
```diff
@@ -1,146 +1,172 @@
...
-			catch(PDOException $e)
-			    {
-			    $sqler =  $sql . "<br>" . $e->getMessage();
```
Comment (mattmolo) [inline complete/completeprocessing.php]: Missing an r in err... :(

**Comment by brstew** [comment]
Any receipt I upload gives a HTTP 500 error. I have the code running on dev-money.purdueieee.org.

**Comment by brstew** [comment]
In PHP 7 you can't use breaks outside of loops or switch statements.

Here is the error message from the php logs.

[Tue Jun 26 10:52:28.842967 2018] [:error] PHP Fatal error:  'break' not in the 'loop' or 'switch' context in /srv/web/dev-money/complete/completeprocessing.php on line 78, referer: https://dev-money.purdueieee.org/complete/index.php

**Comment by mattmolo** [comment]
Umm... Wtf php. I'll fix this in a bit then

**Comment by mattmolo** [comment]
@brstew appeared to fix it. I tested it a bit on the dev server. If you want to run through some test cases please do. Check the 35b374f for the new changes

**Comment by rakoskr6** [comment]
I would probably get rid of adding the username to the file as it's unnecessary (I realized you already merged it). I don't think there were ever any collisions - it was just bad error reporting. 

And yeah... this page was gross. Thanks for the help! Other than e9ee102

**Pull request merged**